src/test/java/hillel/spring/doctor/controller/DoctorControllerTest.java

```diff
@@ -38,154 +24,152 @@
-        when(doctorService.findBySpecialization("surgeon")).thenReturn(List.of(new Doctor(1, "Hide", "surgeon")));
+    public void findDoctorsByNameNotFound() throws Exception {
+        addDoctor(1, "Hide", "dentist");
 
-        this.mockMvc.perform(get("/doctors?specialization={spec}&name={name}", "surgeon", "A")
+        this.mockMvc.perform(get("/doctors?name={name}", "A")
                 .accept(MediaType.APPLICATION_JSON))
                 .andExpect(status().isOk())
-                .andExpect(jsonPath("$", hasSize(1)))
-                .andExpect(jsonPath("$[0].id").value(1))
-                .andExpect(jsonPath("$[0].name").value("Hide"))
-                .andExpect(jsonPath("$[0].specialization").value("surgeon"));
-
-        verify(doctorService, times(1)).findBySpecialization("surgeon");
-        verify(doctorService, times(0)).findByNameStartsWith(anyString());
+                .andExpect(jsonPath("$", hasSize(0)));
     }
 
     @Test
-    public void createDoctorWithPredefinedId() throws Exception {
-        this.mockMvc.perform(post("/doctors")
-                .content("{\"id\": 1, \"name\": \"Hide\", \"specialization\": \"dentist\"}")
-                .contentType(MediaType.APPLICATION_JSON))
-                .andExpect(status().isBadRequest());
+    public void findDoctorsByNameAndSpecialization() throws Exception {
+        addDoctor(1, "Hide", "dentist");
+        addDoctor(2, "Abbott", "surgeon");
+        addDoctor(3, "archibald", "therapist");
+        addDoctor(4, "Abbey", "surgeon");
 
-        verify(doctorService, times(0)).create(any());
+        this.mockMvc.perform(get("/doctors?specialization={spec}&name={name}", "surgeon", "Abb")
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(2)))
+                .andExpect(jsonPath("$[0].id").value(2))
+                .andExpect(jsonPath("$[0].name").value("Abbott"))
+                .andExpect(jsonPath("$[0].specialization").value("surgeon"))
+                .andExpect(jsonPath("$[1].id").value(4))
+                .andExpect(jsonPath("$[1].name").value("Abbey"))
+                .andExpect(jsonPath("$[1].specialization").value("surgeon"));
     }
 
     @Test
     public void createDoctor() throws Exception {
-        //doNothing().when(doctorService).create(any(Doctor.class));
-
         this.mockMvc.perform(post("/doctors")
                 .content("{\"name\": \"Hide\", \"specialization\": \"dentist\"}")
                 .contentType(MediaType.APPLICATION_JSON))
                 .andExpect(status().isCreated());
 
-        verify(doctorService, times(1)).create(new Doctor(null, "Hide", "dentist"));
+        assertEquals(doctorRepository.list().size(), 1);
+
+        Doctor savedDoctor = doctorRepository.list().get(0);
+        assertEquals(savedDoctor.getName(), "Hide");
+        assertEquals(savedDoctor.getSpecialization(), "dentist");
```

Comment:
In Junit assertions, the first  parameter is expected value, and the second one is actual.
And It is  better to use assertions from `AssertJ`